pkg/detectors/ipquality/ipquality.go

```diff
@@ -79,3 +76,51 @@ func (s Scanner) Type() detectorspb.DetectorType {
 func (s Scanner) Description() string {
 	return "IPQualityScore provides tools to detect and prevent fraudulent activity. IPQualityScore API keys can be used to access their fraud prevention services."
 }
+
+func verifyIPQualityAPIKey(ctx context.Context, client *http.Client, apiKey string) (bool, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://www.ipqualityscore.com/api/json/account/%s", apiKey), nil)
+	if err != nil {
+		return false, err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var response apiResponse
+
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return false, err
+		}
+
+		if err = json.Unmarshal(bodyBytes, &response); err != nil {
+			return false, err
+		}
+
+		switch response.Success {
+		case true:
+			return true, nil
+		case false:
+			if strings.Contains(response.Message, insufficientCreditMessage) {
```

Comment:
Updating the logic to report keys as not live if the response message indicates "insufficient credit."  

**Reasoning:**  
- Deleted or incorrect keys also return this message.  
- Even if the API key exists, it remains unusable due to insufficient credit.  

To minimize false positives, it's more accurate to classify keys as not live in this scenario.

Reply:
Well maybe we can also add something in metadata to inform user about this 🤔

Reply:
Added 🚀